src/solver/modeler/api/include/antares/solver/modeler/api/linearProblemBuilder.h

```diff
@@ -24,17 +24,18 @@
 #include <vector>
 
 #include "linearProblemFiller.h"
-#include "mipSolution.h"
 
 namespace Antares::Solver::Modeler::Api
 {
 
 class LinearProblemBuilder
 {
 public:
-    virtual void LinearProblemBuilder(std::vector<LinearProblemFiller*> fillers) = 0;
-    virtual void build(LinearProblemData* data) = 0;
-    virtual MipSolution* solve() = 0;
+    LinearProblemBuilder(const std::vector<std::shared_ptr<LinearProblemFiller>>& fillers);
```

Comment:
the use of shared_ptr is questionnable

Reply:
Could you be more specific ? What would you suggest ?

Reply:
if the ownership LinearProblemFiller is managed externally , you could simply use raw pointer or if you which to ensure non-nullity of LinearProblemFiller use ref

Reply:
| Feature                     | `shared_ptr<LinearProblemFiller>`      | `LinearProblemFiller*`               | `LinearProblemFiller&`                |
|-----------------------------|----------------------------------------|--------------------------------------|---------------------------------------|
| **Memory Management**        | Automatic (via reference counting)     | Manual (requires `new`/`delete`)     | Manual (no ownership semantics)       |
| **Ownership**                | Shared ownership                      | No ownership                         | No ownership                          |
| **Null-Safety**              | Can be null                           | Can be null                          | Cannot be null                        |
| **Memory Overhead**          | Extra overhead for reference counting | No overhead                          | No overhead                           |
| **Lifetime Guarantee**       | Managed, object destroyed when last owner releases | External management                 | External management                   |
| **Preferred Use Case**       | When multiple owners exist or unsure of lifetime management | When you know the lifetime management outside the builder | When fillers are guaranteed to outlive the builder |

Reply:
the use new/delete is not mandatory, for ex if LinearProblemFiller is obtained from a unique_ptr::get()

Reply:
ok. Done.

Reply:
It was also done when handling the LP (linear problem) inside the linear problem fillers